flang-rt/lib/flang_rt/CUDA/CMakeLists.txt

```diff
@@ -0,0 +1,34 @@
+#===-- lib/flang_rt/CUDA/CMakeLists.txt ------------------------------------===#
+#
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+#===------------------------------------------------------------------------===#
+
+
+add_flangrt_library(CufRuntime STATIC
```

Comment:
Why the change to STATIC only [here](https://github.com/llvm/llvm-project/pull/121782/commits/323504f8160921684e0c26543df6c8b1203bdc13)?

We also currently build and test the share version in our current CUDA Fortran feature development (@clementval).

Can you add SHARED in https://github.com/llvm/llvm-project/pull/121782?

Reply:
Because this:
https://github.com/llvm/llvm-project/blob/main/flang/cmake/modules/AddFlangOffloadRuntime.cmake#L19-L23

Do you build `FLANG_RT_EXPERIMENTAL_OFFLOAD_SUPPORT=CUDA` and `FLANG_CUF_RUNTIME` independently? Not yet one of the configurations I have tested.

Reply:
> Do you build FLANG_RT_EXPERIMENTAL_OFFLOAD_SUPPORT=CUDA and FLANG_CUF_RUNTIME=ON independently? 

Yes, the Cuda Fortra runtime does not need (and I think should not?) be built with `FLANG_RT_EXPERIMENTAL_OFFLOAD_SUPPORT=CUDA` option. @clementval is the expert that can confirm here.

Reply:
The CUDA Fortran runtime is (at least for now) host only entry points. So it is not called from the device so @jeanPerier is right. This is independent of `FLANG_RT_EXPERIMENTAL_OFFLOAD_SUPPORT` and should be buildable for static or shared.

Reply:
Added support for shared CufRuntime to #121782 in 6c03632ed7ad254a59d29ede5072b900ed0d0191